### PR TITLE
Add unit test for kubelet pod attach: TestAttach

### PR DESCRIPTION
**Pull request by zhangxiaoyu-zidif**
**What this PR does / why we need it**:

Add unit test: TestAttach

**Special notes for your reviewer**:
It's difficult for me to finish Indirect streaming case now.
I may do it in another PR later.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/kubelet_pods_test.go [hunk 1]
```diff
@@ -1753,6 +1753,94 @@ func TestExec(t *testing.T) {
...
+		expectError: true,
+	}, {
+		description: "no such container",
+		podFullName: podFullName,
+		container:   "containerBar",
+		expectError: true,
+	}}
+
+	for _, tc := range testcases {
+		testKubelet := newTestKubelet(t, false /* controllerAttachDetachEnabled */)
```
Comment (mtaufen) [inline pkg/kubelet/kubelet_pods_test.go]: prefer `//` at end of line rather than comment interleaved with function call

**Review by mtaufen — COMMENTED** [review]
lgtm other than one minor nit
Thanks for your work!

**Comment by mtaufen** [comment]
@k8s-bot ok to test

**Comment by mtaufen** [comment]
You can also remove the release note section from the PR description.

**Comment by zhangxiaoyu-zidif** [comment]
@mtaufen  I fix the comment and others by the way =)

**Review by MHBauer — APPROVED** [review]
not sure about the comment changes but the test looks good.

**Comment by zhangxiaoyu-zidif** [comment]
ping @vishh

**Comment by YuPengZTE** [comment]
LGTM

**Comment by zhangxiaoyu-zidif** [comment]
ping @Random-Liu

**Comment by zhangxiaoyu-zidif** [comment]
@wojtek-t , do you mind approving it？

**Comment by zhangxiaoyu-zidif** [comment]
@k8s-bot pull-kubernetes-unit test this

**Comment by zhangxiaoyu-zidif** [comment]
ping @xiangpengzhao 
/ok-to-test

**Comment by xiangpengzhao** [comment]
Stephen, please fix this failure.
```
W0814 13:29:11.717] Use --sandbox_debug to see verbose messages from the sandbox.
W0814 13:29:11.719] 2017/08/14 13:29:11 error running compiler: exit status 1
W0814 13:29:11.719] Use --strategy=GoCompile=standalone to disable sandboxing for the failing actions.
I0814 13:29:11.821] pkg/kubelet/kubelet_pods_test.go:1830: undefined: podWithUidNameNs
W0814 13:29:12.121] ____Building complete.
W0814 13:29:12.171] ____Elapsed time: 112.448s, Critical Path: 31.25s
I0814 13:29:12.272] //hack:verify-boilerplate                                                PASSED in 23.7s
I0814 13:29:12.273] 
I0814 13:29:12.274] Executed 1 out of 2 tests: 1 test passes and 1 fails to build.
W0814 13:29:12.378] Run: ('/workspace/./test-infra/jenkins/../scenarios/../images/pull_kubernetes_bazel/coalesce.py',)
I0814 13:29:12.993] Build failed
E0814 13:29:13.007] Command failed
I0814 13:29:13.010] process 469 exited with code 1 after 35.3m
E0814 13:29:13.012] FAIL: pull-kubernetes-bazel
```

Note that `pkg/kubelet/kubelet_pods_test.go:1830: undefined: podWithUidNameNs`.

It caused the unit and bazel failures.

**Comment by zhangxiaoyu-zidif** [comment]
OK, thanks. I will fix it soon. :0）

**Pull request closed without merging**